tgui/packages/tgui/interfaces/common/sorting/SortableTable/type.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { InfernoNode } from "inferno";
+import { SortDirection } from "../type";
+
+/** Configuration for Sortable Table header row and callback functions */
+export interface SortableTableHeaderConfig {
+
+  /** Presentation of header cell for this field */
+  children: InfernoNode,
+
+  /** Is this field sortable? */
+  sortable: BooleanLike,
+
+  /** Is this field searchable */
+  searchable: BooleanLike,
```

Comment:
```suggestion
  /** Is this field sortable? */
  sortable?: boolean,

  /** Is this field searchable */
  searchable?: boolean,
```
`BooleanLike`s are a tgui concept for how boolean-ish data gets given to the component via backend state (i.e. `ui_data` proc), as those are *actually* given as `1` and `0` but should be handled as booleans rather than numbers with the caveats involved in doing so.

Essentially, if the items are defined in the UI code, you're making `boolean`s with real `true` and `false` values. Also, no harm in making these optional if you then handle non-existence as "not searchable" or "not sortable", would help keep the config a little leaner.